framework/Source/Operations/CBRescaleEffect.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  CBRescaleEffect.swift
+//
+//
+//  Created by Chiaote Ni on 2024/5/2.
+//
+
+import Foundation
+import Metal
+
+public class CBRescaleEffect: BasicOperation {
+    // 1 ~ 4000
+    public var targetDimension: Int = 1024
+
+    public init() {
+        super.init(fragmentFunctionName: "passthroughFragment", numberOfInputs: 1)
+    }
+
+    public override func newTextureAvailable(_ texture: Texture, fromSourceIndex: UInt) {
+        guard fromSourceIndex == 0 else {
+            assertionFailure("sourceIndex out of range")
+            return
+        }
+        guard let commandBuffer = sharedMetalRenderingDevice.commandQueue.makeCommandBuffer() else {
+            return
+        }
+
+        defer {
+            textureInputSemaphore.signal()
+        }
+        let _ = textureInputSemaphore.wait(timeout:DispatchTime.distantFuture)
```

Comment:
suggest to change the order
```suggestion
        let _ = textureInputSemaphore.wait(timeout: DispatchTime.distantFuture)
        defer {
            textureInputSemaphore.signal()
        }
```